components/infobox/wikis/mobilelegends/infobox_hero_custom.lua

```diff
@@ -0,0 +1,180 @@
+local Cell = require('Module:Infobox/Widget/Cell')
+local Center = require('Module:Infobox/Widget/Center')
+local Title = require('Module:Infobox/Widget/Title')
+local Header = require('Module:Infobox/Widget/Header')
+local Breakdown = require('Module:Infobox/Widget/Breakdown')
+
+local CustomHero = Class.new()
+
+local CustomInjector = Class.new(Injector)
+
+local _args
+
+local _pagename = mw.title.getCurrentTitle().text
+local _frame
+
+local _BATTLE_POINTS_ICON = '[[File:Mobile_Legends_BP_icon.png|x16px|Battle Points|link=Battle Points]]'
+local _DIAMONDS_ICON = '[[File:Mobile_Legends_Diamond_icon.png|Diamonds|x16px|link=Diamonds]]'
+
+function CustomHero.run(frame)
+	local unit = Unit(frame)
+	_args = unit.args
+	_frame = frame
+	_args.informationType = 'Hero'
+
+	unit.getWikiCategories = CustomHero.getWikiCategories
+	unit.setLpdbData = CustomHero.setLpdbData
+	unit.createWidgetInjector = CustomHero.createWidgetInjector
+
+	return unit:createInfobox(frame)
+end
+
+function CustomInjector:addCustomCells()
+	local widgets = {
+		Cell{name = 'Resource Bar', content = {_args.secondarybar}},
+		Cell{name = 'Secondary Bar', content = {_args.secondarybar1}},
+		Cell{name = 'Secondary Attributes', content = {_args.secondaryattributes1}},
+		Cell{name = 'Release Date', content = {_args.releasedate}},
+		Cell{name = 'Species', content = {_args.species}},
+		Cell{name = 'Year of birth', content = {_args.birth}},
+		Cell{name = 'Faction(s)', content = {_args.factions}},
+	}
+
+	if not (
+		String.isEmpty(_args.hp) and
+		String.isEmpty(_args.hplvl) and
+		String.isEmpty(_args.hpreg) and
+		String.isEmpty(_args.hpreglvl)
+	) then
+		table.insert(widgets, Title{name = 'Base Statistics'})
+	end
+
+	table.insert(widgets, Cell{name = 'Health', content = {_args.hp}})
+	table.insert(widgets, Cell{name = 'Health Regen', content = {_args.hpreg}})
+	table.insert(widgets, Cell{name = 'Mana', content = {_args.mana}})
+	table.insert(widgets, Cell{name = 'Mana Regen', content = {_args.manareg}})
+	table.insert(widgets, Cell{name = 'Cooldown Reduction', content = {_args.cdr}})
+	table.insert(widgets, Cell{name = 'Energy', content = {_args.energy}})
+	table.insert(widgets, Cell{name = 'Energy Regen', content = {_args.energyreg}})
+	table.insert(widgets, Cell{name = 'Attack Type', content = {_args.attacktype}})
+	table.insert(widgets, Cell{name = 'Attack Damage', content = {_args.damage}})
+	table.insert(widgets, Cell{name = 'Attack Speed', content = {_args.attackspeed}})
+	table.insert(widgets, Cell{name = 'Attack Range', content = {_args.attackrange}})
+	table.insert(widgets, Cell{name = 'Ability Power', content = {_args.ap}})
+	table.insert(widgets, Cell{name = 'Armor', content = {_args.armor}})
+	table.insert(widgets, Cell{name = 'Magic Resistance', content = {_args.magicresistance}})
+	table.insert(widgets, Cell{name = 'Movement Speed', content = {_args.movespeed}})
+	table.insert(widgets, Title{name = 'Esports Statistics'})
```

Comment:
This feels as a very similar list to that of Infobox Item? I'm thinking the data should probably go into a third, shared, module?

Reply:
if items and heroes share such params and cells it should definitely use a shared source for those

Reply:
okay checked them and they share some params but not all, both have additional ones
the order is different and the display too, so not easy to use a shared source for them

Reply:
I'm guessing that order isn't important?
Wikicode parameter names can be adjusted/standardised, it's a brand new wiki after all.
Isn't the difference in display mostly that Hero uses the raw value, while Item parsed through functions at first?

Reply:
Resolving this convo in fair of https://github.com/Liquipedia/Lua-Modules/pull/714#discussion_r794423716